save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/utils/FileUtils.kt

```diff
@@ -4,9 +4,54 @@
 
 package com.saveourtool.save.agent.utils
 
+import com.saveourtool.save.agent.fs
+import com.saveourtool.save.core.files.findAllFilesMatching
 import okio.FileNotFoundException
 import okio.FileSystem
+import okio.Path
 import okio.Path.Companion.toPath
+import platform.posix.S_IRGRP
+import platform.posix.S_IROTH
+import platform.posix.S_IRUSR
+import platform.posix.S_IWUSR
+import platform.posix.S_IXUSR
+
+/**
+ * Extract path as ZIP archive to provided directory
+ *
+ * @param targetPath
+ */
+internal fun Path.extractZipTo(targetPath: Path) {
+    require(fs.metadata(targetPath).isDirectory)
+    logDebugCustom("Unzip ${fs.canonicalize(this)} into ${fs.canonicalize(targetPath)}")
+    platform.posix.system("unzip $this -d $targetPath")
+}
+
+/**
+ * Write content of [this] into a file [file]
+ *
+ * @receiver [ByteArray] to be written into a file
+ * @param file target [Path]
+ * @param mustCreate will be passed to Okio's [FileSystem.write]
+ */
+internal fun ByteArray.writeToFile(file: Path, mustCreate: Boolean = true) {
+    fs.write(
+        file = file,
+        mustCreate = mustCreate,
+    ) {
+        write(this@writeToFile).flush()
+    }
+}
+
+/**
+ * Mark [this] file as executable. Sets permissions to rwxr--r--
+ */
+internal fun Path.markAsExecutable() {
+    platform.posix.chmod(
+        this.toString(),
+        (S_IRUSR or S_IWUSR or S_IXUSR or S_IRGRP or S_IROTH).toUInt()
```

Comment:
such S_ prefixes should have a warning in diktat :)

Reply:
These are imported C symbols, and diktat checks only declarations